internal/framework/client/client.go

```diff
@@ -0,0 +1,39 @@
+package client
```

Comment:
This code looks pretty similar to what we have in the `internal/provider` package, would it be possible to colocate it there?

Reply:
Yes, we could move it there. Initially it was doing more things that included Framework only types so I created a separate package. However, this is only intended for resources implemented with the TF Plugin Framework so I kept it in the `internal/framework` package for that reason. I thought it would be good to have a package that contains framework-only client helpers. I don't feel too strongly about it either way though.